pyproject.toml

```diff
@@ -317,6 +317,7 @@ lint.ignore = [
     # https://beta.ruff.rs/docs/rules/#flake8-pyi-pyi
     "PYI053", # string-or-bytes-too-long - causes mypy to fail on some of our type stubs
     "PYI054", # numeric-literal-too-long - causes mypy to fail on some of our type stubs
+    "UP035", # TODO: remove once min version of pydantic supports using collections.abc types
```

Comment:
This caused issues with our min version of pydantic when using things like `collections.abc.Mapping` instead of `typing.Mapping`.